File: Python/questions/BestTimeToBuyAndSellStockII/best-time-to-buy-and-sell-stock-ii.py
def maxProfit(prices: list[int]) -> int:
    result = 0
    
    for i in range(1, len(prices)):
        prev, curr = prices[i - 1], prices[i]
        
        if curr > prev:
            result += curr - prev
            
    return result

if __name__ == "__main__":
    print(maxProfit([7,1,5,3,6,4]))
    print(maxProfit([1,2,3,4,5]))
    print(maxProfit([7,6,4,3,1]))

File: Java/src/questions/BestTimeToBuyAndSellStockII/BestTimeToBuyAndSellStockII.java
public class BestTimeToBuyAndSellStockII {
    public static int maxProfit(int[] prices) {
        int result = 0;

        for (int i = 1; i < prices.length; i++) {
            int diff = prices[i] - prices[i - 1];
            if (diff > 0) {
                result += diff;
            }
        }

        return result;
    }

    public static void main(String[] args) {
        System.out.println(BestTimeToBuyAndSellStockII.maxProfit(new int[] { 7, 1, 5, 3, 6, 4 }));
        System.out.println(BestTimeToBuyAndSellStockII.maxProfit(new int[] { 1, 2, 3, 4, 5 }));
        System.out.println(BestTimeToBuyAndSellStockII.maxProfit(new int[] { 7, 6, 4, 3, 1 }));
    }
}
